me.id.webverify/webverifylib/src/main/java/me/id/webverifylib/IDmeWebVerify.java

```diff
@@ -365,12 +373,24 @@ private String createRegisterAffiliationUrl(IDmeAffiliationType affiliationType)
    * @return URL: with proper formatted request
    */
   private String createRegisterConnectionUrl(IDmeConnectionType connectionType, IDmeScope scope) {
-    String url = idMeWebVerifyGetAuthUri
-        .replace(CLIENT_ID_KEY, clientId)
-        .replace(REDIRECT_URI_KEY, redirectUri)
-        .replace(RESPONSE_TYPE_KEY, RESPONSE_TYPE_VALUE)
-        .replace(SCOPE_TYPE_KEY, scope.getScopeId());
-    return String.format("%s&connect=%s", url, connectionType.getKey());
+    return getCommonUri()
+        .appendQueryParameter(SCOPE_TYPE_KEY, scope.getScopeId())
+        .appendQueryParameter(CONNECT_TYPE_KEY, connectionType.getKey())
+        .build()
+        .toString();
+  }
+
+  /**
+   * Creates the common URL which contains the client id, the client, the redirect uri and the response type
+   */
+  private Uri.Builder getCommonUri() {
+    return new Uri.Builder()
+        .scheme(idMeWebVerifyGetCommonUri.getScheme())
```

Comment:
if the order doesn't matter when building the uri, maybe you can sort this alphabetically or maybe set these values in the order they have in the final url (first scheme, then host, then path, then params)